### DISCUSSION TITLE
Bindr - How does it work?

### DISCUSSION THREAD

**Discussion by KorcakDaniel** [hyprwm/Hyprland#5033] [Q&A] [answered]
Hi, I have this binding set up, but for some reason only the first block gets executed.

```bindr=, XF86Keyboard, exec, brightnessctl --device='tpacpi::kbd_backlight' set 0 || brightnessctl --device='tpacpi::kbd_backlight' set 1 || brightnessctl --device='tpacpi::kbd_backlight' set 2 ```

According to the wiki's Bind flags section it should work.
https://wiki.hyprland.org/Configuring/Binds/#bind-flags

Anything I'm doing wrong?

**Comment by vaxerski** [comment]
1) wdym first block
2) uhhh 
![image](https://github.com/hyprwm/Hyprland/assets/43317083/e9b1d331-62a9-4659-badd-d9c292681fcd)

**Reply by KorcakDaniel** [reply]
Sorry, I was checking older versions for clues, and forgot to switch. I've edited the orginal question with a correct link.

**Reply by KorcakDaniel** [reply]
What I meant by block is the described functionality of bindr in bind flags, where you can map different operations onto one key depending on the number of presses. In this example I'm trying to map 3 levels of a keyboard backlight - off, low, max (0, 1, 2).

**Comment by KorcakDaniel** [comment]
Bump

**Comment by vaxerski** [comment]
why should the other blocks be executed? `r` just means release, so instead of activating on key DOWN it will activate on key UP. `brightnessctl --device='tpacpi::kbd_backlight' set 0` never fails so only this gets executed. Basic bash.

**Reply by KorcakDaniel** [reply]
I see, I misunderstood the functionality of bindr then. The example suggests something else:

```
# Start wofi opens wofi on FIRST press, closes it on SECOND
bindr=SUPER, SUPER_L, exec, pkill wofi || wofi
```

Thank you for clarification.

**Reply by vaxerski** [reply, marked as answer]
this doesn't suggest anything like that. `pkill` returns false if the process is not found, and thus this will launch wofi if it's not open and close it if it is.

The `r` there is to bind a single modkey

**Reply by KorcakDaniel** [reply]
I see now, thank you. Therefore I can make a simple bash script which turns off the backlight, returns false if it isn't lit and then turn it on. Very much appreciated. :)
